### PR TITLE
fix(build): proper prod pypi deploys

### PR DESCRIPTION
**Pull request by sfoster1**
This should now correctly clear the OT_BUILD environ variable for the
pypi upload by collapsing the two steps into one and using a local
change to the env var rather than writing the env file

### CODE REVIEW

File: .github/actions/python/pypi-deploy/action.yaml [hunk 1]
```diff
@@ -20,16 +20,12 @@ runs:
...
                  echo "::error ::Invalid repository url ${{ inputs.repository_url }}"
                  exit 1
              fi
         fi
-
-
-    - shell: bash
-      run: |
         status=0
         QUIET=1 BUILD_NUMBER=${OT_BUILD} make -C ${{ inputs.project }} clean deploy twine_repository_url=${{ inputs.repository_url }} pypi_username=opentrons pypi_password=${{ inputs.password }} || status=$?
```
Comment (mcous) [inline .github/actions/python/pypi-deploy/action.yaml:31]: Should we move the arguments to be after `make` to match the other arguments?

File: .github/actions/python/pypi-deploy/action.yaml [hunk 2]
```diff
@@ -20,16 +20,12 @@ runs:
         else if [[ ${{ inputs.repository_url }} =~ "upload.pypi.org" ]]; then
                  echo "Uploading to prod pypi"
```
Comment (mcous) [inline .github/actions/python/pypi-deploy/action.yaml]: Should we fix up the indentation while we're here?

File: .github/actions/python/pypi-deploy/action.yaml [hunk 3]
```diff
@@ -20,16 +20,12 @@ runs:
...
                  # this needs to be cleared otherwise the makefiles append a dev tag
-                 echo 'OT_BUILD=' >> $GITHUB_ENV
+                 OT_BUILD=
```
Comment (mcous) [inline .github/actions/python/pypi-deploy/action.yaml]: I wonder if this logic could be moved to the Makefile somehow. The fact that it's so difficult to test is a bit of a warning bell to me (acknowledging that GH actions doesn't really make anything easy to test).

For example, the app-shell uses the presence of `OT_TAG` to drive the answer to the question "is this a prob build or not?", which feels a little easier to follow than the Python behavior of checking for a non-OT-prefixed `BUILD_NUMBER` env var

**Review by mcous — APPROVED** [review]
Looks to me like this will work (?), but left some thoughts that went through my head when I was examining this code last week

File: .github/actions/python/pypi-deploy/action.yaml (hunk 3 above)
Comment (sfoster1) [reply .github/actions/python/pypi-deploy/action.yaml]: I would like to make this better yeah but honestly I'm not sure where the right place is for some of this stuff. Having it all in a big inscrutable makefile doesn't really feel any better to me, at first glance, than in a script here.

File: .github/actions/python/pypi-deploy/action.yaml (hunk 2 above)
Comment (sfoster1) [reply .github/actions/python/pypi-deploy/action.yaml]: you're not into 8-space tabs?

**Pull request merged**